Modules/_threadmodule.c

```diff
@@ -39,6 +40,127 @@ get_thread_state(PyObject *module)
     return (thread_module_state *)state;
 }
 
+// _ThreadHandle type
+
+typedef struct {
+    PyObject_HEAD
+    unsigned long long ident;
+    Py_uintptr_t handle;
+    char joinable;
```

Comment:
I admit this is slightly seducive. I'll let that idea bake in my mind a bit and come back to it later :-)

Reply:
Ok, this opaque `PyThread_handle_t` probably wouldn't work precisely because it is opaque to the outside, so it would need to be dynamically-allocated inside the PyThread APIs. That starts looking like a lot of pointless work.

Reply:
You may be right.  It's not something to dig into all that far in this PR.